### PR TITLE
Added annotatevariants loci

### PR DESCRIPTION
**Pull request by tpoterba**
- Added a new annotate module due to demand from users to be able to annotate variants without ref/alt.
 - Added `Locus` constructor and accessor methods in the expr language

### CODE REVIEW

**Comment by tpoterba** [comment]
jenkins

File: src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsLoci.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    if (files.isEmpty)
+      fatal("Arguments referred to no files")
+
+    val vds = state.vds
+
+    val (expr, code) = (Option(options.code), Option(options.root)) match {
+      case (Some(c), None) => (true, c)
+      case (None, Some(r)) => (false, r)
+      case _ => fatal("this module requires one of `--root' or `--code', but not both")
+    }
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsLoci.scala]: I think `--root` should be deprecated.  I wouldn't add it for new commands.

File: src/main/scala/org/broadinstitute/hail/expr/Type.scala [hunk 2]
```diff
@@ -564,6 +572,18 @@ case class TStruct(fields: IndexedSeq[Field]) extends Type {
     filter(fn)
   }
 
+  def queryFields[T](code: String, expected: Type): (Annotation) => Option[T] = {
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/expr/Type.scala]: Don't like the name.  How about `parseInStructScope`?

**Comment by cseed** [comment]
Looks great.  Back to you.

File: src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsLoci.scala (hunk 1 above)
Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsLoci.scala]: I think we should keep things consistent for now.  I can make another PR to add deprecation warnings for all --root args, but I'd like this module to have similar usage to `annotatevariants table`

Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsLoci.scala]: You know I'm a sucker for the consistency argument.  Sounds like a plan.

**Comment by tpoterba** [comment]
Changed queryFields to parseInStructScope.  Back to you.

**Comment by tpoterba** [comment]
jenkins

**Comment by tpoterba** [comment]
jenkins

**Pull request merged**